samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -19,11 +19,28 @@ const ShareView = () => {
     const [uShowCount, uShowSetCount] = useState(0);
 
     // Declare new state variables that are required to disable the submit vote button.
-    const [disabled, setDisabled] = useState(false)
+    const [isVoteBtnDisabled, setIsVoteBtnDisabled] = useState(false);
 
-    useEffect(() => {
-        microsoftTeams.initialize();
+    // Declare new state variables that are required to set the username of users who sign in using SSO too.
+    const [userName, setUserName] = useState("");
+
+    // Declare new state variables that are required to disable the Facebook button after login
+    const [IsdisabledfacebookButton, setIsdisabledfacebookButton] = useState(true);
+
+    // Declare new state variables that are required for login success load the submit vote button
+    const [enableVoteDiv, setEnableVoteDiv] = useState(false);
+
+    // Declare new state variables that are required for a verified anonymous user or a normal user
+    const [IsAnonymousUser, setIsAnonymousUser] = useState(true);
+
+    // Declare new state variables that are required for a verified anonymous user or a normal user
+    const [IsVisibleConsent, setIsVisibleConsent] = useState(false);
+
+    // Declare new state variables that are required disable the authentication button after login
+    const [ssoAuthenticationBtn, setSsoAuthenticationBtn] = useState(true);
 
+    useEffect(() => {
+        microsoftTeams.app.initialize();
     }, [])
```

Comment:
Do we actually need this? Or is the SDK being initialised in router.js good enough? 
Similarly for the auth-start/auth-ends, you may be able to remove them there too.

I'm not sure there is anything wrong with duplicate initialise per say but it just feels incorrect.

Reply:
Fixed